### DISCUSSION TITLE
Using both getInitialProps and getStaticProps

### DISCUSSION THREAD

**Discussion by doptster** [vercel/next.js#18004] [Help]
Currently, it seems that it's unavoidable for my app for use _app.getInitialProps for user data, but on the page's perspective, I think I could optimized the api call by using getStaticProps (with revalidate). What I want to know is, is the combination a good practice and doable solution?

**Comment by mrclay** [comment]
Should a static page be served if users are logged out?
Can you just activate user features client side? Or would they need to receive custom markup?

**Comment by mrclay** [comment]
Preview Mode makes getStaticProps() dynamic for some users. Worth looking into especially if you want static content for those logged out.
